primitives/src/trie/trie_proof_node.rs

```diff
@@ -58,6 +59,29 @@ impl From<TrieNode> for TrieProofNode {
 }
 
 impl TrieProofNode {
+    pub fn include_value(node: TrieNode) -> Self {
+        let value = match (node.has_children(), node.value) {
+            (_, None) => panic!("can't include value of a node without one"),
+            (false, Some(val)) => ProofValue::LeafValue(val),
+            (true, Some(val)) => ProofValue::HybridValue(val),
+        };
+        TrieProofNode {
+            key: node.key,
+            value,
+            children: node.children,
+        }
+    }
+}
+
+impl TrieProofNode {
+    pub fn into_value(self) -> Option<Vec<u8>> {
+        match self.value {
+            ProofValue::None => None,
+            ProofValue::LeafValue(value) => Some(value),
+            ProofValue::HybridHash(_) => None,
+            ProofValue::HybridValue(value) => Some(value),
+        }
+    }
```

Comment:
Might it make more sense to not take self but use and return references?

```suggestion
    pub fn value(&self) -> Option<&Vec<u8>> {
        match &self.value {
            ProofValue::None => None,
            ProofValue::LeafValue(value) => Some(value),
            ProofValue::HybridHash(_) => None,
            ProofValue::HybridValue(value) => Some(value),
        }
    }
```